src/disposable/AsyncDisposable.ts

```diff
@@ -0,0 +1,42 @@
+import { IAsyncDisposable } from './types';
```

Comment:
Nitpick - file should be name `asyncDisposable.ts` to be aligned with `Disposable`

Reply:
Nitpick - all the files are named wrong can I fix the casing?

Reply:
yeah